packages/menu-button/src/index.tsx

```diff
@@ -136,7 +141,13 @@ export const Menu: React.FC<MenuProps> = ({ id, children }) => {
       // @ts-ignore
       window.__REACH_DISABLE_TOOLTIPS = true;
       window.requestAnimationFrame(() => {
-        focus(menuRef.current);
+        if (keepScrollPosition) {
+          const scrollOptions = { left: window.scrollX, top: window.scrollY };
+          focus(menuRef.current);
+          window.scrollTo(scrollOptions);
+        } else {
+          focus(menuRef.current);
+        }
```

Comment:
I think we could also clean this up slightly by only having the scrollTo behind a conditional

```js
const scrollOptions = { left: window.scrollX, top: window.scrollY };

focus(menuRef.current);

if (keepScrollPosition) {
  window.scrollTo(scrollOptions);
}
```